### PR TITLE
feat: support read throttling by size

### PR DESCRIPTION
**Pull request by Smityz**
Ref: https://github.com/apache/incubator-pegasus/issues/830

### CODE REVIEW

File: src/server/pegasus_server_impl.cpp [hunk 1]
```diff
@@ -53,6 +54,12 @@ DSN_DEFINE_int32("pegasus.server",
                  10,
                  "hotkey analyse interval in seconds");
 
+DSN_DEFINE_int32("pegasus.server",
+                 read_throttling_by_size_request_count,
+                 1,
+                 "the default value of throttling controller of read_throttling_by_size");
```
Comment (Shuo-Jia) [inline src/server/pegasus_server_impl.cpp:60]: the config seem to be useless, which case need change the config?

**Review by levy5307 — DISMISSED** [review]

File: src/server/pegasus_server_impl.cpp [hunk 2]
```diff
@@ -274,6 +281,12 @@ void pegasus_server_impl::on_get(get_rpc rpc)
     resp.partition_index = _gpid.get_partition_index();
     resp.server = _primary_address;
 
+    if (!_read_size_throttling_controller->get_token(FLAGS_read_throttling_by_size_request_count)) {
```
Comment (Shuo-Jia) [inline src/server/pegasus_server_impl.cpp:284]: exactly should getToken(rpc.size)

File: src/server/pegasus_server_impl.cpp [hunk 3]
```diff
@@ -274,6 +281,12 @@ void pegasus_server_impl::on_get(get_rpc rpc)
...
+        rpc.error() = dsn::ERR_BUSY;
+        _counter_recent_read_throttling_reject_count->increment();
```
Comment (Shuo-Jia) [inline src/server/pegasus_server_impl.cpp:286]: I suggest the reject count can be distinct the `size` and `qps`, which we can get the request why exceed the throld

File: src/server/pegasus_server_impl.cpp (hunk 2 above)
Comment (Smityz) [reply src/server/pegasus_server_impl.cpp:284]: Actually, it has nothing to do with any size, it only checks if the token basket is empty. If it's empty, then reject all requests

Comment (Shuo-Jia) [reply src/server/pegasus_server_impl.cpp:284]: You have define the default value in get_token, no need pass the FLAGS_read_throttling_by_size_request_count

Comment (Smityz) [reply src/server/pegasus_server_impl.cpp:284]: it can be remotely changed if necessary

File: src/server/pegasus_server_impl.cpp (hunk 3 above)
Comment (Smityz) [reply src/server/pegasus_server_impl.cpp:286]: ref https://github.com/XiaoMi/rdsn/blob/master/src/replica/replica_throttle.cpp#L56
For compatibility,  I think it's ok to combine them. We can distinguish `size` and `qps` from other perf_counters easily

File: src/server/pegasus_server_impl.cpp (hunk 2 above)
Comment (Shuo-Jia) [reply src/server/pegasus_server_impl.cpp:284]: You just get current the avaliable token, so which case need change? or why change

File: src/server/pegasus_server_impl.cpp (hunk 3 above)
Comment (Shuo-Jia) [reply src/server/pegasus_server_impl.cpp:286]: Next pr consider it, now is ok

File: src/base/pegasus_const.h [hunk 4]
```diff
@@ -68,4 +68,6 @@ extern const std::string ROCKSDB_BLOCK_CACHE_ENABLED;
 extern const std::string SPLIT_VALIDATE_PARTITION_HASH;
```
Comment (levy5307) [inline src/base/pegasus_const.h:68]: Could you add some tests for this pull request?

Comment (Smityz) [reply src/base/pegasus_const.h:68]: I will add some function tests for all read/write throttle tests in my next PR

**Review by levy5307 — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**